core/src/main/scala/zio/jdbc/SqlFragment.scala

```diff
@@ -113,27 +113,28 @@ sealed trait SqlFragment { self =>
     foreachSegment { syntax =>
       sql.append(syntax.value)
     } { param =>
+      var size = 0
       param.value match {
         case iterable: Iterable[_] =>
-          iterable.iterator.foreach { item =>
+          iterable.foreach { item =>
             paramsBuilder += item.toString
+            size += 1
           }
-          sql.append(
-            Seq.fill(iterable.iterator.size)("?").mkString(",")
-          )
 
         case array: Array[_] =>
           array.foreach { item =>
             paramsBuilder += item.toString
+            size += 1
           }
-          sql.append(
-            Seq.fill(array.length)("?").mkString(",")
-          )
 
         case _ =>
-          sql.append("?")
           paramsBuilder += param.value.toString
+          size += 1
       }
+      sql.append(
+        if (size == 1) "?"
```

Comment:
The if is not needed, is it? `mkstring` won't apply if size is 1 anyway.

Reply:
You are right. `mkString` will handle a single-item case, but it will require additional allocations, which we can easily drop with the `if` condition. I assume that a multi-item placeholder is much more rare than a single-item placeholder. So, I think this "micro" optimization is excused.